x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_edit_rules.ts

```diff
@@ -64,61 +54,5 @@ export const bulkEditRules = async ({
     },
   });
 
-  // rulesClient bulkEdit currently doesn't support bulk mute/unmute.
-  // this is a workaround to mitigate this,
-  // until https://github.com/elastic/kibana/issues/139084 is resolved
-  // if rule actions has been applied:
-  // - we go through each rule
-  // - mute/unmute if needed, refetch rule
-  // calling mute for rule needed only when rule was unmuted before and throttle value is  NOTIFICATION_THROTTLE_NO_ACTIONS
-  // calling unmute needed only if rule was muted and throttle value is not NOTIFICATION_THROTTLE_NO_ACTIONS
-  const ruleActions = attributesActions.filter((rule): rule is BulkActionEditPayloadRuleActions =>
-    [BulkActionEditType.set_rule_actions, BulkActionEditType.add_rule_actions].includes(rule.type)
-  );
-
-  // bulk edit actions are applying in a historical order.
-  // So, we need to find a rule action that will be applied the last, to be able to check if rule should be muted/unmuted
-  const rulesAction = ruleActions.pop();
-
-  if (rulesAction) {
-    const muteOrUnmuteErrors: BulkEditError[] = [];
-    const rulesToMuteOrUnmute = await pMap(
-      result.rules,
-      async (rule) => {
-        try {
-          if (rule.muteAll && rulesAction.value.throttle !== NOTIFICATION_THROTTLE_NO_ACTIONS) {
-            await rulesClient.unmuteAll({ id: rule.id });
-            return (await readRules({ rulesClient, id: rule.id, ruleId: undefined })) ?? rule;
-          } else if (
```

Comment:
For visibility. We've discussed on a sync that this logic should remain here so users can unmute muted rules when bulk adding actions.